jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Callback.java

```diff
@@ -202,13 +203,37 @@ public void completed()
      */
     static Callback from(InvocationType invocationType, Runnable completed)
     {
-        return new Completing(invocationType)
+        return new Completing()
         {
             @Override
             public void completed()
             {
                 completed.run();
             }
+
+            @Override
+            public void succeeded()
```

Comment:
It feels like `succeeded()` and `completed()` should have default implementations calling `completed()` in the interface.